Add page maker app

baru64 opened the pull request.

baru64 committed:
Merge pull request #7 from Artemis21/tameTNT-app-maker

tameTNT-app-maker

baru64 committed:
Add basic ModelForms

baru64 committed:
Add auth views and urls

Changes:
  - added basic auth, base and main page templates
  - added urls for auth and main page
  - added generic auth views and simple main view

Artemis21 committed:
Enforce PEP8

Artemis21 committed:
Merge pull request #15 from Artemis21/add-auth-views

Add auth views and urls

baru64 committed:
Add user view classes and templates

baru64 committed:
Add webpage and template views - work in progress

baru64 committed:
Fix static files

baru64 committed:
Add webpage and template views, urls and base templates

baru64 committed:
Add comment views

Changes:
  - add comment section to webpage-detail
  - add comments to WebpageDetailView context
  - fix CommentCreate and WebpageCreate success url
  - add LOGIN_URL to settings

A5rocks committed:
Test logging in and the home view

While not much, this test can act like a template for others.
Here's some notes on the tests for my later perusal:
 - `client.<method>(<relative url>).context` will tell you what was
   passed into the returned template. This way, we don't have to
   hard-code html into our tests!

 - `client.login(username=<username>, password=<password>)` just
   sets the authentication cookie (or something?) Then
   `client.logout()` will remove said authentication. Seems useful!

 - cookies persist through calls automatically!

 - CSRF protection is disabled - no need to inspect a form or
   something cumbersome to do like that.

 - `django.tests.TestCase` is a subclassed `unittest.TestCase`,
   which lets you use your fancy `self.assertIn`, `self.assertIs`,
   etc.

baru64 committed:
Fix WebpageUpdate, WebpageDelete and CommentDelete

WebpageUpdateView and WebpageDeleteView had missing
slug_field, slug_url_kwarg and success_url.
Added comment delete form to webpage_detail template.
Added success_url for CommentDeleteView.

baru64 committed:
Add TemplateDetailView, TemplateListView

Changes:
  - TemplateDetailView, TemplateListView
  - Added header block to base template to insert css
    from WebpageView and TemplateDetailView

A5rocks committed:
Make a second test for templates

I also tried making a test around page creation / commenting, but I
couldn't get Django to figure out an image was an image...
Hopefully I find some article about doing this before the end of
the code jam...

baru64 committed:
Split tests, add users fixture

baru64 committed:
Merge pull request #19 from Artemis21/add-view-tests

Add simple tests for UserRegister and Template views

baru64 committed:
Add UserViewsTest

baru64 committed:
Add WebpageViewsTest, TemplateViewsTest, CommentViewsTest

baru64 committed:
Fix CommentCreateView redirect

baru64 committed:
Fix test_comment_delete

baru64 committed:
Improve tests

Changes:
  - remove trailing slash in urls
  - swap hardcoded paths with reverse()
  - remove template tests

baru64 committed:
Merge pull request #25 from Artemis21/add-more-tests

Add more tests

baru64 committed:
Merge pull request #27 from Artemis21/add-base-views

Add base views

baru64 committed:
Stylistic changes, add placeholder image

A5rocks committed:
Merge branch 'master' into add-page_maker-app

Artemis21 commented on jolly-jellyfish/src/page_maker/fixtures/templates.yaml:
```diff
@@ -0,0 +1,14 @@
+    name: hotdog
+    date_created: 2020-08-02 16:22:00+00:00
+    author: 1
+    style_sheet: styles/test.css
```
Would it make more sense to have one of the included themes here?

A5rocks replied:
Pretty sure these files don't exist to begin with:

https://github.com/Artemis21/summer-code-jam-2020/pull/25#discussion_r467231618

Artemis21 replied:
Oh, ok. @baru64 can you explain why?

baru64 replied:
We only test the behavior of view, what objects does it return, what is the status code and when we check if there is an image or file in a page we only check if there is a link to it, we don't check if you can download the image.
To use actual files we can change the MEDIA_ROOT for tests to something like /test_media/ but there's not point in this unless we want to test downloading.

Artemis21 replied:
Oh, of course, I thought this was what we discussed about including some articles default.

Artemis21 commented on jolly-jellyfish/src/page_maker/fixtures/templates.yaml:
```diff
@@ -0,0 +1,14 @@
+    name: doghot
+    date_created: 2020-08-02 19:51:03.603472+00:00
+    author: 1
+    style_sheet: styles/test2.css
```
As above.

Artemis21 commented on jolly-jellyfish/src/page_maker/forms.py:
```diff
@@ -0,0 +1,36 @@
+
+    class Meta:
+        model = Template
+        fields = ['name', 'style_sheet']
```
Would be cool if we could have a nicer widget for style_sheet, maybe with previews of the styles. Probably for another PR though.

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/base.html:
```diff
@@ -0,0 +1,41 @@
+    </header>
+
+    <main>
+      {% if error %}
+      <div>
+        <p>{{ error.text }}</p>
+      </div>
+      {% endif %}
```
What sort of errors would these be? Could they be moved to after the content? Styling concerns...

baru64 replied:
I just dropped it in before creating views, but I will remove it since it's not used.

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/template_create.html:
```diff
@@ -0,0 +1,13 @@
+{% extends "page_maker/base.html" %}
+
+{% block content %}
+<div>
```
```suggestion
{% block content %}
<h2>Create a Template</h2>
<div>
```
Style finds first element and calls it a header.

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/template_delete.html:
```diff
@@ -0,0 +1,14 @@
+{% extends "page_maker/base.html" %}
+
+{% block content %}
+<div>
```
```suggestion
{% block content %}
<h2>Delete Template</h2>
<div>
```
As above, styling concerns.

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/template_list.html:
```diff
@@ -0,0 +1,11 @@
+{% extends "page_maker/base.html" %}
+
+{% block content %}
+{% for template in templates %}
```
```suggestion
{% block content %}
<h2>Templates</h2>
{% for template in templates %}
```
As above, styling concerns.

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/user_delete.html:
```diff
@@ -0,0 +1,14 @@
+{% extends "page_maker/base.html" %}
+
+{% block content %}
+<div>
```
```suggestion
{% block content %}
<h2>Delete Acount</h2>
<div>
```
As above, styling concerns.

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/user_detail.html:
```diff
@@ -0,0 +1,25 @@
+{% extends "page_maker/base.html" %}
+
+{% block content %}
+<div>
+  username: {{ viewed_user.username }}
```
```suggestion
{% block content %}
username: {{ viewed_user.username }}
<div>
```
As above, styling concerns.

EDIT: Don't use this, see below.

A5rocks replied:
Shouldn't there be a header too?

Artemis21 replied:
Oh, you're right
```suggestion
{% block content %}
<h2>Username: {{ viewed_user.username }}</h2>
<div>
```

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/user_update.html:
```diff
@@ -0,0 +1,13 @@
+{% extends "page_maker/base.html" %}
+
+{% block content %}
+<div>
```
```suggestion
{% block content %}
<h2>Update Account</h2>
<div>
```
As above, styling concerns.

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/webpage_create.html:
```diff
@@ -0,0 +1,13 @@
+{% extends "page_maker/base.html" %}
+
+{% block content %}
+<div>
```
```suggestion
{% block content %}
<h2>Create Page</h2>
<div>
```
As above, styling concerns.

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/webpage_delete.html:
```diff
@@ -0,0 +1,14 @@
+{% extends "page_maker/base.html" %}
+
+{% block content %}
+<div>
```
```suggestion
{% block content %}
<h2>Delete Page</h2>
<div>
```
As above, styling concerns.

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/webpage_list.html:
```diff
@@ -0,0 +1,16 @@
+{% extends "page_maker/base.html" %}
+
+{% block content %}
+{% for webpage in webpages %}
```
```suggestion
{% block content %}
<h2>Top Pages</h2>
{% for webpage in webpages %}
```
As above, styling concerns.

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/webpage_update.html:
```diff
@@ -0,0 +1,13 @@
+{% extends "page_maker/base.html" %}
+
+{% block content %}
+<div>
```
```suggestion
{% block content %}
<h2>Update Page</h2>
<div>
```
As above, styling concerns.

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/registration/login.html:
```diff
@@ -0,0 +1,24 @@
+{% extends "page_maker/base.html" %}
+
+{% block content %}
+    {% if form.errors %}
```
```suggestion
{% block content %}
    <h2>Login</h2>
    {% if form.errors %}
```
As above, styling concerns.

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/registration/register.html:
```diff
@@ -0,0 +1,17 @@
+{% extends "page_maker/base.html" %}
+{% block content %}
+   <div>
```
```suggestion
{% block content %}
   <h2>Sign up</h2>
   <div>
```
As above, styling concerns.

Artemis21 commented on jolly-jellyfish/src/page_maker/tests.py:
```diff
@@ -1,3 +1,230 @@
+        self.assertEqual(user.email, 'foo@example.com')
+
+    def test_register_and_home(self):
+        # assumptions (aka what this tests):
+        #  - `/user/register` exists.
+        #  - it takes certain params.
+        #  - it also logs us in.
+        #  - `/` exists.
+        #  - it tells us our username.
+        #  - we use django's built-in auth system.
```
Might be better written as a docstring?

Artemis21 commented on jolly-jellyfish/src/page_maker/tests.py:
```diff
@@ -1,3 +1,230 @@
+        # we create an account with the `USERNAME` and `PASSWORD` vars
+        params = {
+            'username': USERNAME,
+            'email': 'foo@example.com',
```
For consistency, email should probably be a constant also.

Artemis21 commented on jolly-jellyfish/src/page_maker/tests.py:
```diff
@@ -1,3 +1,230 @@
+        self.assertEqual(User.objects.count(), 1)
+        user = User.objects.get(pk=1)
+        self.assertEqual(user.username, USERNAME)
+        self.assertEqual(user.email, 'foo@example.com')
```
See above, email as a constant?

Artemis21 commented on jolly-jellyfish/src/page_maker/tests.py:
```diff
@@ -1,3 +1,230 @@
+        self.client.login(username=USERNAME, password=PASSWORD)
+
+    def test_template_create(self):
+        # what this assumes:
+        # - when you create a page the
+        #   available templates are
+        #   passed in. and in a certain
+        #   way.
+        # - posting `/templates/new`
+        #   creates a specified template
```
Perhaps better written as a docstring?

Artemis21 commented on jolly-jellyfish/src/page_maker/views.py:
```diff
@@ -1,3 +1,242 @@
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        # user = get_object_or_404(User, pk=self.kwargs.get('pk'))
```
Useless code?

Artemis21 commented on jolly-jellyfish/src/page_maker/views.py:
```diff
@@ -1,3 +1,242 @@
+class WebpageCreateView(LoginRequiredMixin, FormView):
+    template_name = 'page_maker/webpage_create.html'
+    form_class = WebpageForm
+    # success_url = '/pages'
```
Useless code?

Artemis21 commented on jolly-jellyfish/src/page_maker/views.py:
```diff
@@ -1,3 +1,242 @@
+    form_class = WebpageForm
+    # success_url = '/pages'
+
+    # TODO validate and sanitize text and images
```
How much of this does Django do for us?

baru64 replied:
It probably handles sanitizing both text and images, I removed the comment.

Artemis21 commented on jolly-jellyfish/src/page_maker/views.py:
```diff
@@ -1,3 +1,242 @@
+        pagename = self.kwargs.get('pagename')
+        return reverse_lazy('webpage-detail', kwargs={'pagename': pagename})
+    
+    # redirect to detail view
```
Better written inside the function/as a docstring?

Artemis21 reviewed (changes requested):
Looks good. Several suggestions adding titles to pages, mainly for styling.

A5rocks commented on jolly-jellyfish/src/page_maker/fixtures/users.yaml:
```diff
@@ -0,0 +1,30 @@
+- model: auth.user
+  pk: 1
+  fields: # password: hard123passwd
```
Hm maybe we want to doublespace this comment?

A5rocks commented on jolly-jellyfish/src/page_maker/fixtures/users.yaml:
```diff
@@ -0,0 +1,30 @@
+    user_permissions: []
+- model: auth.user
+  pk: 2
+  fields: # password: foobar < xyzzy
```
Same as above

A5rocks commented on jolly-jellyfish/src/page_maker/forms.py:
```diff
@@ -0,0 +1,36 @@
+
+
+class CommentForm(forms.ModelForm):
+
```
An unnecessary linebreak?

A5rocks commented on jolly-jellyfish/src/page_maker/forms.py:
```diff
@@ -0,0 +1,36 @@
+
+
+class WebpageForm(forms.ModelForm):
+
```
Same as above?

A5rocks commented on jolly-jellyfish/src/page_maker/forms.py:
```diff
@@ -0,0 +1,36 @@
+
+
+class TemplateForm(forms.ModelForm):
+
```
Same as above...

A5rocks commented on jolly-jellyfish/src/page_maker/templates/page_maker/base.html:
```diff
@@ -0,0 +1,41 @@
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+
+    {% if webpage %}
+      <title>{{ webpage.user_title }}</title>
+    {% else %}
+      <title>Early Internet Page Maker</title>
+    {% endif %}
```
Pretty sure we aren't indenting inside if blocks...

Artemis21 replied:
Maybe we should though

A5rocks commented on jolly-jellyfish/src/page_maker/templates/page_maker/base.html:
```diff
@@ -0,0 +1,41 @@
+  <body>
+    <header>
+      Hello!
+      {% if user.is_authenticated %}
+        <a href="{% url 'user-detail' pk=user.id %}"> {{ user.username }}</a>,
+        <a href="{% url 'logout' %}">logout</a>
+      {% else %}
+        <a href="{% url 'login' %}">Login</a>
+        <a href="{% url 'register' %}">Register</a>
+      {% endif %}
```
Same as above

A5rocks commented on jolly-jellyfish/src/page_maker/templates/page_maker/webpage_detail.html:
```diff
@@ -0,0 +1,30 @@
+<div>
+  <form method="POST" enctype="multipart/form-data" action="{% url 'comment-create' pagename=webpage.name %}">
+    {% csrf_token %}
+      <div>
+        {{ comment_form.as_p }}
+        <button type="submit">Add comment</button>
+      </div>
+    </form>
```
Extra indent.

A5rocks commented on jolly-jellyfish/src/page_maker/templates/registration/login.html:
```diff
@@ -0,0 +1,24 @@
+{% extends "page_maker/base.html" %}
+
+{% block content %}
+    {% if form.errors %}
+        <p>Your username and password didn't match. Please try again.</p>
+    {% endif %}
+
+    <form method="post" action="{% url 'login' %}">
+    {% csrf_token %}
+         <table>
+         <tr>
+             <td>{{ form.username.label_tag }}</td>
+             <td>{{ form.username }}</td>
+         </tr>
+         <tr>
+             <td>{{ form.password.label_tag }}</td>
+             <td>{{ form.password }}</td>
+         </tr>
+         </table>
+ 
+         <input type="submit" value="login" />
+         <input type="hidden" name="next" value="{{ next }}" />
+     </form>
+{% endblock %}
```
Weirdly indented? As in, everything inside the block should have 1 level less of indentation.

A5rocks commented on jolly-jellyfish/src/page_maker/templates/registration/login.html:
```diff
@@ -0,0 +1,24 @@
+{% extends "page_maker/base.html" %}
+
+{% block content %}
+    {% if form.errors %}
+        <p>Your username and password didn't match. Please try again.</p>
+    {% endif %}
```
This probably shouldn't be indented extra.

A5rocks commented on jolly-jellyfish/src/page_maker/templates/registration/login.html:
```diff
@@ -0,0 +1,24 @@
+
+    <form method="post" action="{% url 'login' %}">
+    {% csrf_token %}
+         <table>
+         <tr>
+             <td>{{ form.username.label_tag }}</td>
+             <td>{{ form.username }}</td>
+         </tr>
+         <tr>
+             <td>{{ form.password.label_tag }}</td>
+             <td>{{ form.password }}</td>
+         </tr>
+         </table>
+ 
+         <input type="submit" value="login" />
+         <input type="hidden" name="next" value="{{ next }}" />
+     </form>
```
All these are indented with 1 space too much :O

... What am I doing with my life to be scanning the space count like that in a couple places :///

A5rocks commented on jolly-jellyfish/src/page_maker/templates/registration/register.html:
```diff
@@ -0,0 +1,17 @@
+{% extends "page_maker/base.html" %}
+{% block content %}
+   <div>
+       <form method="POST">
+           {% csrf_token %}
+            <div>
+                {{ form.as_p }}
+                <button type="submit"> Sign Up </button>
+            </div>
+        </form>
+    <div>
+        <small>
+            Already Have An Account?  <a href="{% url 'login' %}"> Sign In </a>
+        </small>
+    </div>
+    </div>
+{% endblock content %}
```
Weird indentation? As in: there's a level of indentation that shouldn't be there in the content block.

A5rocks commented on jolly-jellyfish/src/page_maker/templates/registration/register.html:
```diff
@@ -0,0 +1,17 @@
+{% block content %}
+   <div>
+       <form method="POST">
+           {% csrf_token %}
```
In most places we have this line 1 indent lower (to be the same level as the form before)

A5rocks reviewed (changes requested):
I very much skimmed through this one (as Artemis has already gone through and I've gone through the constituent PRs), just pointed out a couple things I found.

Two things:
  1) We'll want to format html at the end for consistency's sake.
  2) This will be a PAIN to resolve merge conflicts with tameTNT's PR... You and him took different directions a bunch.

Artemis21 commented:
As @A5rocks has pointed out, it would be good to lint/beautify the html, and perhaps also the yaml. Does anyone know of good linters for django-html or yaml?

baru64 committed:
Apply styling suggestions

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/registration/login.html:
```diff
@@ -0,0 +1,23 @@
+{% extends "page_maker/base.html" %}
+
+{% block content %}
+{% if form.errors %}
```
```suggestion
{% block content %}
<h2>Log in</h2>
{% if form.errors %}
```
Still missing a header.

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/registration/register.html:
```diff
@@ -0,0 +1,17 @@
+{% extends "page_maker/base.html" %}
+{% block content %}
+<div>
```
```suggestion
{% block content %}
<h2>Sign Up</h2>
<div>
```
As above.

Artemis21 reviewed (commented):
Header still missing in a couple of files.

baru64 committed:
Add missing headers

A5rocks commented on jolly-jellyfish/src/page_maker/templates/registration/login.html:
```diff
@@ -1,24 +1,24 @@
+<form method="post" action="{% url 'login' %}">
+{% csrf_token %}
+  <table>
+  <tr>
+    <td>{{ form.username.label_tag }}</td>
+    <td>{{ form.username }}</td>
+  </tr>
+  <tr>
+    <td>{{ form.password.label_tag }}</td>
+    <td>{{ form.password }}</td>
+  </tr>
```
```suggestion
    <tr>
      <td>{{ form.username.label_tag }}</td>
      <td>{{ form.username }}</td>
    </tr>
    <tr>
      <td>{{ form.password.label_tag }}</td>
      <td>{{ form.password }}</td>
    </tr>
```

very minor change though

A5rocks commented on jolly-jellyfish/src/page_maker/tests.py:
```diff
@@ -163,17 +167,20 @@ def setUp(self):
         # example stylesheet
-        style = io.StringIO('body { display: hidden; }')
         template_name = 'blahblahblah'
+        style = str(Path(settings.BASE_DIR) / 'static' / 'themes' / 'console.css')
```
We might want to revert this change.

A5rocks reviewed (commented):
LGTM, just a really minor nitpick and something that we should make a separate issue for

A5rocks commented:
These are probably beyond the scope of the PR:
 - https://github.com/Artemis21/summer-code-jam-2020/pull/28#discussion_r467342421
 - https://github.com/Artemis21/summer-code-jam-2020/pull/28#discussion_r467483310

Let's put them into another PR so we can finally merge this one :^)

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/template_create.html:
```diff
@@ -0,0 +1,14 @@
+<h2>Create a template</h2>
+<div>
+  <form method="POST" enctype="multipart/form-data">
+  {% csrf_token %}
```
Should this line be further indented?

A5rocks replied:
I guess? We'll have to edit tameTNT's PR to have this too though.

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/template_delete.html:
```diff
@@ -0,0 +1,15 @@
+<h2>Delete a template</h2>
+<div>
+  <form method="POST">
+  {% csrf_token %}
```
As above

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/user_delete.html:
```diff
@@ -0,0 +1,15 @@
+<h2>Delete an account</h2>
+<div>
+  <form method="POST">
+  {% csrf_token %}
```
As above

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/user_update.html:
```diff
@@ -0,0 +1,14 @@
+<h2>Update an account</h2>
+<div>
+  <form method="POST">
+  {% csrf_token %}
```
As above.

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/webpage_create.html:
```diff
@@ -0,0 +1,14 @@
+    <div>
+      {{ form.as_p }}
+      <button type="submit">Create page</button>
+    </div>
```
What is the purpose of this div?

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/webpage_delete.html:
```diff
@@ -0,0 +1,15 @@
+<h2>Delete a webpage</h2>
+<div>
+  <form method="POST">
+  {% csrf_token %}
```
Should this line be further indented?

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/webpage_delete.html:
```diff
@@ -0,0 +1,15 @@
+      {{ form.as_p }}
+      <p>Delete page <b>{{webpage.name}}?</b></p>
+      <button type="submit">Confirm</button>
+    </div>
```
What is the purpose of this div?

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/webpage_detail.html:
```diff
@@ -0,0 +1,30 @@
+{% extends "page_maker/base.html" %}
+
+{% block content %}
+<div><h1>{{ webpage.user_title }}</h1></div>
```
```suggestion
<h1><span><span>{{ webpage.user_title }}</span></span></h1>
```
This will have the custom theme, right?

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/webpage_detail.html:
```diff
@@ -0,0 +1,30 @@
+    <div>
+      {{ comment_form.as_p }}
+      <button type="submit">Add comment</button>
+    </div>
```
Unnecessary div?

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/webpage_update.html:
```diff
@@ -0,0 +1,14 @@
+<h2>Update a webpage</h2>
+<div>
+  <form method="POST">
+  {% csrf_token %}
```
Indent?

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/page_maker/webpage_update.html:
```diff
@@ -0,0 +1,14 @@
+    <div>
+      {{ form.as_p }}
+      <button type="submit">Update</button>
+    </div>
```
Unnecessary div?

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/registration/login.html:
```diff
@@ -0,0 +1,24 @@
+{% endif %}
+
+<form method="post" action="{% url 'login' %}">
+{% csrf_token %}
```
Indent?

Artemis21 commented on jolly-jellyfish/src/page_maker/templates/registration/register.html:
```diff
@@ -0,0 +1,18 @@
+    <div>
+      {{ form.as_p }}
+      <button type="submit"> Sign Up </button>
+    </div>
```
Unnecessary div?

Artemis21 commented on jolly-jellyfish/src/page_maker/tests.py:
```diff
@@ -1,3 +1,237 @@
+
+
+class UserRegisterTest(TestCase):
+    def setUp(self):
```
This name seems out of style, or is it required by something else?

A5rocks replied:
It's required by `unittest`.

baru64 committed:
Remove divs from templates

A5rocks committed:
More formatting of the HTML

A5rocks committed:
Fix for standard templates

The standard templates that ship with the app require doubly nested
spans (for spacing reasons...), so we need to make sure to supply these
for the Django templates.
